src/pytorch_lightning/cli.py

```diff
@@ -393,13 +394,14 @@ def init_parser(self, **kwargs: Any) -> LightningArgumentParser:
         )
         return parser
 
-    def setup_parser(
```

Comment:
Removing these arguments would be considered a breaking change. Can you add a deprecation cycle for them?

Reply:
I oversaw this. Why is `setup_parser` public? In which use cases is necessary to override it? I can change it so that we keep `_setup_parser_kwargs` and `setup_parser` still receives `main_kwargs` and `subparser_kwargs`.

Reply:
Reverted the removal of _setup_parser_kwargs method and setup_parser parameters.